base_layer/chat_ffi/src/lib.rs

```diff
@@ -0,0 +1,153 @@
+// following disclaimer in the documentation and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote
+// products derived from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+// WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#![recursion_limit = "1024"]
+
+use std::{
+    ffi::{CStr, CString},
+    fs::File,
+    io::Read,
+    path::PathBuf,
+    str::FromStr,
+};
+
+use libc::c_char;
+use log::{debug, warn};
+use tari_chat_client::Client;
+use tari_common::configuration::Network;
+use tari_common_types::tari_address::TariAddress;
+use tari_comms::{peer_manager::Peer, NodeIdentity};
+use tari_p2p::P2pConfig;
+use tari_shutdown::Shutdown;
+use tokio::runtime::Runtime;
+
+pub struct ClientFFI {
+    client: Client,
+    runtime: Runtime,
+    shutdown: Shutdown,
+}
+
+const LOG_TARGET: &str = "chat_ffi";
+
+/// Creates a Chat Client
+///
+/// ## Arguments
+/// `config` - The P2PConfig pointer
+/// `identity_file_path` - The path to the node identity file
+/// `db_path` - The path to the db file
+/// `network_str` - The network to connect to
+/// ## Returns
+/// `*mut ChatClient` - Returns a pointer to a ChatClient, note that it returns ptr::null_mut()
+/// if config is null, an error was encountered or if the runtime could not be created
+///
+/// # Safety
+/// The ```client_destroy``` method must be called when finished with a TariWallet to prevent a memory leak
+#[no_mangle]
+pub unsafe extern "C" fn create_chat_client(
+    config: *mut P2pConfig,
+    identity_file_path: *const c_char,
+    db_path: *const c_char,
+    seed_peers: *mut *mut Peer,
+    network_str: *const c_char,
+) -> *mut ClientFFI {
+    let identity_path = PathBuf::from(
+        CStr::from_ptr(identity_file_path)
+            .to_str()
+            .expect("A non-null identity path should be able to convert to a string"),
+    );
+    let mut buf = Vec::new();
+    File::open(identity_path)
+        .expect("Can't open the identity file")
+        .read_to_end(&mut buf)
+        .expect("Can't read the identity file into buffer");
```

Comment:
can you try and make then return some error code rather than crashing on an except?